File: src/main/java/com/github/wakingrufus/mastodon/ui/feeds/TootController.java
package com.github.wakingrufus.mastodon.ui.feeds;

import com.github.wakingrufus.mastodon.ui.FittedWebView;
import com.github.wakingrufus.mastodon.ui.ViewAccountKt;
import com.sys1yagi.mastodon4j.api.entity.Status;
import javafx.fxml.FXML;
import javafx.scene.control.Label;
import javafx.scene.layout.HBox;
import lombok.extern.slf4j.Slf4j;

@Slf4j
public class TootController {
    private final Status status;
    @FXML
    HBox accountView;
    @FXML
    FittedWebView content;

    public TootController(Status status) {
        this.status = status;
    }

    @FXML
    public void initialize() {
        ViewAccountKt.viewAccount(accountView,status.getAccount());
        content.setContent(status.getContent());
        content.setHtmlStylesheetLocation(getClass().getResource("/css/toot-content.css").toString());
    }


}

File: src/main/kotlin/com/github/wakingrufus/mastodon/ui/controllers/TootController.kt
package com.github.wakingrufus.mastodon.ui.controllers

import com.github.wakingrufus.mastodon.account.AccountState
import com.github.wakingrufus.mastodon.toot.boostToot
import com.github.wakingrufus.mastodon.toot.unboostToot
import com.github.wakingrufus.mastodon.ui.Controller
import com.github.wakingrufus.mastodon.ui.FittedWebView
import com.github.wakingrufus.mastodon.ui.Viewer
import com.github.wakingrufus.mastodon.ui.ViewerMode
import com.sys1yagi.mastodon4j.api.entity.Account
import com.sys1yagi.mastodon4j.api.entity.Status
import javafx.fxml.FXML
import javafx.scene.control.Button
import javafx.scene.layout.HBox
import mu.KLogging

class TootController(private val status: Status,
                     private val accountViewer: Viewer<Account> = Viewer(
                             controller = { item -> AccountController(item) },
                             template = "/account.fxml"),
                     private val accountPrompter: () -> (AccountState?)) : Controller<Status> {
    companion object : KLogging()

    @FXML
    internal var accountView: HBox? = null

    @FXML
    internal var content: FittedWebView? = null

    @FXML
    internal var favButton: Button? = null

    @FXML
    internal var boostButton: Button? = null

    @FXML
    override fun initialize() {
        accountViewer.view(parent = accountView!!, item = status.account!!, mode = ViewerMode.APPEND)
        content!!.setContent(status.content)
        content!!.setHtmlStylesheetLocation(javaClass.getResource("/css/toot-content.css").toString())
        if (status.isFavourited) {
            favButton!!.text = "★"
        } else {
            favButton!!.text = "☆"
        }
        if (status.isReblogged) {
            boostButton!!.text = "♻"
        } else {
            boostButton!!.text = "♲"
        }
        boostButton?.setOnAction { actionEvent ->
            val account: AccountState? = accountPrompter.invoke()
            logger.info { "account chosen: $account" }
            if (account != null) {
                if (status.isReblogged) {
                    boostButton!!.text = "♲"
                    unboostToot(id = status.id, client = account.client)
                } else {
                    boostButton!!.text = "♻"
                    boostToot(id = status.id, client = account.client)
                }
            }
        }
    }
}